nucliadb_vectors/src/data_point/mod.rs

```diff
@@ -103,6 +107,96 @@ impl Address {
     }
 }
 
+pub struct FormulaFilter<'a> {
+    pub matching_nodes: HashSet<u64>,
+    empty_formula: bool,
+    filter: &'a Formula,
+    pub use_fst: bool,
+    total_nodes: usize,
+}
+
+impl FormulaFilter<'_> {
+    pub fn new<'a>(
+        filter: &'a Formula,
+        key_index: Option<&'a KeyIndex>,
+        label_index: Option<&'a LabelIndex>,
+        total_nodes: usize,
+    ) -> FormulaFilter<'a> {
+        let matching_nodes: HashSet<u64>;
+        let empty_formula: bool;
+        let use_fst: bool;
+
+        if key_index.is_none() && label_index.is_none() {
+            empty_formula = true;
+            matching_nodes = HashSet::new();
+            use_fst = false;
+        } else {
```

Comment:
Instead of checking with an if, and unwrapping in the `else` branch I think it would be cleaner to use:
```rust
let (Some(key_index), Some(label_index)) = (key_index, label_index) else {
         return FormulaFilter {
                filter,
                total_nodes,
                matching_nodes: HashSet::new(),
                empty_formula: true,
                use_fst: false,
         };
};
```
Less indentation on the rest of the code and no unwraps